pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -358,9 +369,21 @@ message LinuxUser {
     repeated int64 additional_gids = 3;
 }
 
-message ContainerConfig {
-    // Name of the container.
+// ContainerMetadata holds all necessary information for building unique
+// container name.
+message ContainerMetadata {
+    // The name of the container. Same as the container name in the PodSpec.
     optional string name = 1;
+    // The restart count of the container.
+    optional uint32 restart_count = 2;
```

Comment:
This is OK, but "attempt" seems like a better name.  Do we care if this is a monotonic counter or a random number?